ejp_xml_pipeline/etl.py

```diff
@@ -80,23 +92,110 @@ def etl_ejp_xml_zip(
+        load_entities_file_to_s3(
+            ejp_xml_data_config,
+            object_key
+        )
 
-        load_entities_file_to_bq(ejp_xml_data_config)
+
+def get_temp_s3_object_name(
+        obj_prefix_in_config: str,
+        original_object_name: str
+):
+    obj_name = obj_prefix_in_config + original_object_name + '.json'
+
+    return obj_name
 
 
-def load_entities_file_to_bq(ejp_xml_load_config: eJPXmlDataConfig):
+def load_entities_file_to_s3(
+        ejp_xml_load_config: eJPXmlDataConfig,
+        original_obj_key
+):
     for entity in ejp_xml_load_config.entity_type_mapping.values():
         if os.path.getsize(entity.get_full_file_location()) > 0:
-            create_or_extend_table_schema(
-                ejp_xml_load_config.gcp_project,
-                ejp_xml_load_config.dataset,
-                entity.table_name,
-                entity.get_full_file_location()
+            obj_key = get_temp_s3_object_name(
+                entity.s3_object_prefix,
+                original_obj_key
+            )
+            upload_file_into_s3(
+                bucket=ejp_xml_load_config.temp_file_s3_bucket,
+                object_key=obj_key,
+                full_file_path=entity.get_full_file_location()
             )
 
-            load_file_into_bq(
-                filename=entity.get_full_file_location(),
-                table_name=entity.table_name,
-                dataset_name=ejp_xml_load_config.dataset,
-                project_name=ejp_xml_load_config.gcp_project
+
+def load_entity_file_to_bq(
+        gcp_project: str,
+        dataset: str,
+        table_name: str,
+        file_path: str
+):
+    if os.path.getsize(file_path) > 0:
+        create_or_extend_table_schema(
+            gcp_project,
+            dataset,
+            table_name,
+            file_path
+        )
+        load_file_into_bq(
+            filename=file_path,
+            table_name=table_name,
+            dataset_name=dataset,
+            project_name=gcp_project
+        )
+
+
+# pylint: disable='too-many-arguments'
+def download_load2bq_cleanup_temp_files(
+        matching_file_metadata_iter, s3_bucket: str,
+        gcp_project: str, dataset: str,
+        bq_table: str, batch_size_limit: int = 5000
+):
+    written_file_row_count = 0
+    s3_objects_written_to_file = []
+    with NamedTemporaryFile() as named_temp_file:
+        tempfile_name = named_temp_file.name
+        with open(tempfile_name, 'a') as writer:
+            for matching_file_metadata, _ in matching_file_metadata_iter:
+                s3_object = matching_file_metadata.get(
+                    named_literals.S3_FILE_METADATA_NAME_KEY
+                )
+                jsonl_string = download_s3_object_as_string(
+                    s3_bucket,
+                    s3_object
+                )
+                writer.write(jsonl_string)
+                written_file_row_count += sum(
+                    map(lambda x: 1 if '\n' in x else 0, jsonl_string)
```

Comment:
since we go through every character, I guess x will be `\n` if it's a line feed. Might be worth moving this to a separate function (`get_line_number_count` or something similar). (Not sure if it's faster to `len(jsonl_string.splitlines())`.)